C/services/core/configuration_manager.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * FogLAMP FogLAMP Configuration management.
+ *
+ * Copyright (c) 2018 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Massimiliano Pinto
+ */
+
+#include <configuration_manager.h>
+#include <rapidjson/writer.h>
+
+using namespace std;
+
+ConfigurationManager *ConfigurationManager::m_instance = 0;
+
+// Constructor
+ConfigurationManager::ConfigurationManager(const string& host,
+					   unsigned short port)
+{
+	m_storage = new StorageClient(host, port);
+}
+
+// Destructor
+ConfigurationManager::~ConfigurationManager()
+{
+	delete m_storage;
+}
+
+/**
+ * Return the singleton instance of the configuration manager
+ */
```

Comment:
Unusual to have a singleton getInstance tat takes arguments - how will the caller know what the arguments are?

Reply:
This will be changed in future commits